### PR TITLE
112 display channel balance on channels view

### PR DESCRIPTION
**Pull request by markettes**

### CODE REVIEW

File: src/Services/LightningService.cs [hunk 1]
```diff
@@ -1188,5 +1190,27 @@ await foreach (var response in closeChannelResult.ResponseStream.ReadAllAsync())
...
+                    {"macaroon", channel.Node.ChannelAdminMacaroon}
+                }, null, default));
+                
+                var chan = result.Channels.First(x => x.ChanId == channel.ChanId);
+
+                var res = new Tuple<long, long>(chan.LocalBalance, chan.RemoteBalance);
+                return res;
+            }
+            //We avoid to stop the method if the peer is already connected
+            catch (RpcException e)
```
[the remark this answers is not in the document]
Comment (markettes) [reply src/Services/LightningService.cs]: Fixed

File: src/Pages/Channels.razor [hunk 2]
```diff
@@ -280,6 +314,20 @@
         ToastService.ShowSuccess("Text copied");
     }
 
+    private double GetPercentageBalance(Channel channel)
+    {
+        try
+        {
+            var result = LightningService.GetChannelBalance(channel).Result;
```
Comment (cvicenti) [inline src/Pages/Channels.razor:321]: Any reason for not using async/await here?

File: src/Pages/Channels.razor [hunk 3]
```diff
@@ -280,6 +292,12 @@
         ToastService.ShowSuccess("Text copied");
     }
 
+    private double GetPercentageBalance(Channel channel)
+    {
+        var result = LightningService.GetChannelBalance(channel).Result;
+        return (result.Item1 / (double)(result.Item2 + result.Item1)) * 100;
```
[the remark this answers is not in the document]
Comment (markettes) [reply src/Pages/Channels.razor]: Done

File: src/Services/LightningService.cs [hunk 4]
```diff
@@ -1188,5 +1190,27 @@ await foreach (var response in closeChannelResult.ResponseStream.ReadAllAsync())
...
+            try
+            {
+                var result = client.Execute(x => x.ListChannels(new ListChannelsRequest(), 
+                    new Metadata {
+                    {"macaroon", channel.Node.ChannelAdminMacaroon}
+                }, null, default));
+                
+                var chan = result.Channels.First(x => x.ChanId == channel.ChanId);
+
+                var res = new Tuple<long, long>(chan.LocalBalance, chan.RemoteBalance);
```
[the remark this answers is not in the document]
Comment (markettes) [reply src/Services/LightningService.cs]: Done

File: src/Pages/Channels.razor (hunk 2 above)
Comment (markettes) [reply src/Pages/Channels.razor]: Nope, I just added it

**Review by Jossec101 — APPROVED** [review]

**Pull request merged**